lms/djangoapps/lms_xblock/mixin.py

```diff
@@ -142,6 +144,62 @@ def _get_user_partition(self, user_partition_id):
 
         raise NoSuchUserPartitionError("could not find a UserPartition with ID [{}]".format(user_partition_id))
 
+    def get_parent_unit(self):
+        """
+        Finds xblock's parent unit if it exists.
+
+        Returns:
+            xblock: Returns the parent unit xblock if it exists.
+            If no parent unit exists, returns None.
+        """
+        xblock = self
+        while xblock:
+            xblock = xblock.get_parent()
+            if xblock is None:
+                return None
+            parent = xblock.get_parent()
+            if parent is None:
+                return None
+            if parent.category == "sequential":
+                return xblock
+
+    def _has_nonsensical_access_settings(self):
+        """
+        Checks if a block's group access settings do not make sense.
+
+        By nonsensical access settings, we mean a component's access
+        settings which contradict the unit level access in that they
+        restrict access to a component in such a way that no one can
+        see it.
+        For example, if unit level access is set to "Verified", and
+        a component within the unit is set to "Audit", that setting
+        does not make sense because the component will not be
+        visible to anybody.
+        Note:  The only time this contradiction can occur is when
+        a component restricts access to the same partition but a
+        different group than the unit.
+
+        Returns:
+            bool: True if the block is a component and its access
+            settings contradict the unit level access.
+        """
+        parent_unit = self.get_parent_unit()
+        if not parent_unit:
+            return False
+
+        unit_group_access = parent_unit.group_access
+        component_group_access = self.group_access
+
+        for user_partition_id, unit_group_ids in unit_group_access.iteritems():
+            try:
+                component_group_ids = component_group_access[user_partition_id]
+                if component_group_ids and unit_group_ids and not set(unit_group_ids).issubset(set(component_group_ids)):
+                    return True
+            except KeyError:
```

Comment:
So, at least for now, I think we do need this try/except block.  To reproduce the error:

  1. Make a new unit with a component
  2. Restrict unit access to verified
  3. Go into the container page and remove access to verified from that component, you should get a key error.

Not sure what we actually want to do in this case, since the component is still restricted to verified learners, but you have the option to mark it as not restricted to anyone which doesn’t actually change anything and throws a KeyError exception.